pymodbus/framer/rtu_framer.py

```diff
@@ -107,13 +102,10 @@ def advanceFrame(self):
         it or determined that it contains an error. It also has to reset the
         current frame header handle
         """
-        try:
-            self._buffer = self._buffer[self._header['len']:]
-        except KeyError:
-            #   Error response, no header len found
-            self.resetFrame()
+
+        self._buffer = self._buffer[self._header['len']:]
```

Comment:
In this pull request `self._header` now always populated with `len` key (as in other framer classes - binary, ascii, socket, etc). That's why try/catch is not required anymore.